### PR TITLE
Silex 2 and Bugsnag 3 support

### PR DESCRIPTION
**Pull request by fortis**
Please review

### CODE REVIEW

**Comment by fortis** [comment]
Probably it makes sense to create another branch (v2.0.0) for Bugsnag 3 and Silex 2

File: src/Bugsnag/Silex/Provider/BugsnagServiceProvider.php [hunk 1]
```diff
@@ -3,77 +3,49 @@
...
-                );
+              if ($cookies = $request->cookies->all()) {
+                  $params['cookies'] = $cookies;
+              }
 
-                if ($session) {
-                    $params["session"] = $session;
+              $app['bugsnag']->registerCallback(
+                function ($report) use ($params, $error) {
+                    $report->setMetaData($params);
```
Comment (GrahamCampbell) [inline src/Bugsnag/Silex/Provider/BugsnagServiceProvider.php:38]: This won't work I'm afraid. This isn't metadata.

File: composer.json [hunk 2]
```diff
@@ -11,9 +11,9 @@
         "homepage": "https://bugsnag.com"
     }],
     "require": {
-        "php": ">=5.3.0",
-        "silex/silex": "^1.2",
-        "bugsnag/bugsnag": "^2.0"
+        "php": ">=5.5.9",
```
Comment (GrahamCampbell) [inline composer.json:14]: Please just go to 5.5.

Comment (GrahamCampbell) [inline composer.json:14]: Unless silex is 5.5.9?

Comment (fortis) [inline composer.json:14]: @GrahamCampbell silex2 requires 5.5.9 https://github.com/silexphp/Silex/blob/master/composer.json#L18

File: src/Bugsnag/Silex/Provider/BugsnagServiceProvider.php (hunk 1 above)
Comment (fortis) [inline src/Bugsnag/Silex/Provider/BugsnagServiceProvider.php:38]: oh i see, ok will research

Comment (GrahamCampbell) [inline src/Bugsnag/Silex/Provider/BugsnagServiceProvider.php:38]: Don't worry about it. I can sort this out. It'll need some heavier changes. That's why we've not done it yet.

**Comment by GrahamCampbell** [comment]
Replaced by https://github.com/bugsnag/bugsnag-silex/pull/5.

**Pull request closed without merging**